dashboard/src/components/Filter/CodeBlock.tsx

```diff
@@ -1,12 +1,108 @@
+import { LiaInfoCircleSolid } from 'react-icons/lia';
+
+import { FormattedMessage } from 'react-intl';
+
+import { cn } from '@/lib/utils';
+
+import { Tooltip, TooltipTrigger, TooltipContent } from '@/components/Tooltip';
+
+import ColoredCircle from '@/components/ColoredCircle/ColoredCircle';
+
 type TCodeBlockProps = {
   code: string;
+  highlightsClassname?: string;
 };
 
-const CodeBlock = ({ code }: TCodeBlockProps): JSX.Element => {
+const CodeBlock = ({
+  code,
+  highlightsClassname,
+}: TCodeBlockProps): JSX.Element => {
+  let highlightCount = 0;
+  let failCount = 0;
+  let errorCount = 0;
+
+  // removes unwanted HTML tags
+  code = code.replace(/(<|>)/gm, match => {
+    match = match.replace(/</, '&lt');
+    match = match.replace(/>/, '&gt');
+    return match;
+  });
+
+  // highlights fails and errors
+  code = code.replace(
+    /^.*((\berror(:|\s*-*[0-9]|))|(\bfailed(\sto|\swith|\b|:|\s*[0-9]))|(fail(\b|[0-9]))).*$/gim,
+    match => {
+      highlightCount++;
+      if (match.search(/fail/i) !== -1) {
+        // ignores .failed files and fail flags
+        if (match.search(/(\.+failed|failed(\s*0)+|[|/]fail\b)/i) !== -1) {
+          return '<span class="text-blue">' + match + '</span>';
+        }
+        failCount++;
+        return '<span class="text-red">' + match + '</span>';
+      } else {
+        // ignores error flags
+        if (match.search(/((0\s*\t*)+error|error"+)/i) !== -1) {
```

Comment:
I think these regex are probably the best case for debuting unit tests in our app, if you feel that you got the time please add it

(you will need to extract them to functions if you wanna test it)

Use vitest

https://vitest.dev/

I would leave the test in the component folder and create a `CodeBlock.test.tsx`